File: basic/src/main/java/com/ding/basic/media/MediaConfig.java
package com.ding.basic.media;

/**
 * @author lijun Lee
 * @desc 广告配置构建
 * @mail jun_li@dingyuegroup.cn
 * @data 2018/3/21 12:16
 */

public class MediaConfig {

    private String userId;

    private String channelCode;

    private int cityCode;

    private String cityName;

    private float latitude;

    private float longitude;

    private MediaConfig() {
    }

    private MediaConfig(Builder builder) {
        this.userId = builder.userId;
        this.channelCode = builder.channelCode;
        this.cityCode = builder.cityCode;
        this.cityName = builder.cityName;
        this.latitude = builder.latitude;
        this.longitude = builder.longitude;
    }

    public static Builder newBuilder() {
        return new Builder();
    }

    public String getUserId() {
        return userId;
    }

    public String getChannelCode() {
        return channelCode;
    }

    public int getCityCode() {
        return cityCode;
    }

    public String getCityName() {
        return cityName;
    }

    public float getLatitude() {
        return latitude;
    }

    public float getLongitude() {
        return longitude;
    }

    public static class Builder {

        private String userId;

        private String channelCode;

        private int cityCode;

        private String cityName;

        private float latitude;

        private float longitude;

        public Builder setUserId(String userId) {
            this.userId = userId;
            return this;
        }

        public Builder setChannelCode(String channelCode) {
            this.channelCode = channelCode;
            return this;
        }

        public Builder setCityCode(int cityCode) {
            this.cityCode = cityCode;
            return this;
        }

        public Builder setCityName(String cityName) {
            this.cityName = cityName;
            return this;
        }

        public Builder setLatitude(float latitude) {
            this.latitude = latitude;
            return this;
        }

        public Builder setLongitude(float longitude) {
            this.longitude = longitude;
            return this;
        }

        public MediaConfig build() {
            return new MediaConfig(this);
        }
    }
}

File: dy-media/src/hasad/java/com/dy/media/MediaConfig.kt
package com.dy.media

import com.dycm_adsdk.PlatformSDK
import com.dycm_adsdk.inter.ConfigInterface
import com.dycm_adsdk.view.NativeView

/**
 * Desc 广告配置 有
 * Author qiantao
 * Mail tao_qian@dingyuegroup.cn
 * Date 2018/6/8 17:03
 */
object MediaConfig {

    fun getConfig(): ConfigInterface? = PlatformSDK.config()

    fun getAdCount(): Int? = PlatformSDK.config()?.adCount

    fun getConfigExpireMinutes(): Int? = PlatformSDK.config()?.configExpireMinutes

    fun getChapter_limit(): Int? = PlatformSDK.config()?.chapter_limit

    fun getRestAd_sec(): Int? = PlatformSDK.config()?.restAd_sec

    fun getSwitch_sec(): Int? = PlatformSDK.config()?.switch_sec

    fun getAdfree_new_user(): Int? = PlatformSDK.config()?.adfree_new_user

    fun setChannel_code(channel_code: String) {
        PlatformSDK.config().setChannel_code(channel_code)
    }

    fun setAd_userid(user_id: String) {
        PlatformSDK.config().setAd_userid(user_id)
    }

    fun setCityName(cityName: String) {
        PlatformSDK.config().setCityName(cityName)
    }

    fun setCityCode(cityCode: Int) {
        PlatformSDK.config().setCityCode(cityCode)
    }

    fun setLatitude(latitude: Float) {
        PlatformSDK.config().setLatitude(latitude)
    }

    fun setLongitude(longitude: Float) {
        PlatformSDK.config().setLongitude(longitude)
    }

    fun setBookShelfGrid(grid: Boolean) {
        PlatformSDK.config().setBookShelfGrid(grid)
    }

    fun setCusTomParam(param: String) {
        PlatformSDK.config().setCusTomParam(param)
    }

    fun getAdSwitch(ad_mark_id: String): Boolean? = PlatformSDK.config()?.getAdSwitch(ad_mark_id)

    fun ExposureToPlugin(nativeView: NativeView) {
        PlatformSDK.config().ExposureToPlugin(nativeView)
    }

    fun setExpandInfo(bookmap: Map<String, String>) {
        PlatformSDK.config().setExpandInfo(bookmap)
    }
}